### PR TITLE
AEM-129 - Set Content Scores as Tags

### PR DESCRIPTION
**Pull request by wrandall22**
This PR is part of the work required for [AEM-129](https://jira.cru.org/browse/AEM-129). Specifically, it will add the content score as a `cq:tag` in addition to the metadata property `score`. This will happen when syncing scores from the content scoring web application found at https://content-scoring.cru.org

### CODE REVIEW

File: core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java [hunk 1]
```diff
@@ -50,12 +55,34 @@ private void updateScore(
...
+        List<Tag> newTags = Lists.newArrayList();
+
+        for (Tag existingTag : existingTags) {
+            if (existingTag.getTagID().startsWith(SCALE_OF_BELIEF_TAG_PREFIX)) {
+                continue;
             }
+            newTags.add(existingTag);
         }
+
+        Tag scoreTag = tagManager.resolve(SCALE_OF_BELIEF_TAG_PREFIX + score);
```
Comment (mattdrees) [inline core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java:81]: someone/something needs to create these tags initially, right? is that going to be done manually?

File: core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java [hunk 2]
```diff
@@ -50,12 +55,34 @@ private void updateScore(
...
                 node.setProperty("cq:lastModified", now);
                 node.setProperty("cq:lastModifiedBy", "scale-of-belief");
-                Session session = resourceResolver.adaptTo(Session.class);
 
-                if (session != null) {
-                    session.save();
-                }
+                TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+                List<Tag> newTags = buildTagsWithScore(contentResource, tagManager, score);
+                tagManager.setTags(contentResource, newTags.toArray(new Tag[0]));
```
Comment (mattdrees) [inline core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java:61]: does `setTags()` internally call `session.save()`?

**Review by mattdrees — COMMENTED** [review]
questions

File: core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java (hunk 2 above)
Comment (wrandall22) [reply core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java:61]: It does.

File: core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java (hunk 1 above)
Comment (wrandall22) [reply core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java:81]: The tags already exist, and I think it was done manually, yes.

**Review by mattdrees — APPROVED** [review]

**Pull request merged**